src/Sarif/Visitors/ReformattingVisitor.cs

```diff
@@ -43,23 +45,129 @@ public override Run VisitRun(Run node)
                 }
             }
 
+            if (node.Results != null)
+            {
+                var results = new List<Result>();
+
+                foreach (Result result in node.Results)
+                {
+                    results.Add(VisitResult(result));
+                }
+
+                node.Results = results;
+            }
+
             return node;
         }
 
-        internal FileData VisitDictionaryValueNullChecked(string key, FileData node)
+        public override PhysicalLocation VisitPhysicalLocation(PhysicalLocation node)
         {
-            Uri uri;
+            // If we have a binary region, there is no work to do
+            if (node.Region == null ||
```

Comment:
You wrote a helper method `Region.IsBinaryRegion` but didn't use it here (and I like this logic better for implementing it).

Reply:
This logical doesn't strictly identify a binary region. It allows the entirely empty region through, for example, which, strictly speaking, is a binary region. I'm going to hold tight on further changes here until we revisit this topic in context of the SARIF TC.

---
In reply to: [202100932](https://github.com/Microsoft/sarif-sdk/pull/942#discussion_r202100932) [](ancestors = 202100932)